docs/api/search.md

```diff
@@ -0,0 +1,140 @@
+            "description": "A name is associated with every search result so a client has something to display.",
+            "examples": [
+                "Web App Development"
+            ]
+        }
+    },
+    "required": [
+        "id",
+        "name"
+    ]
+}
+```
+
+### Links
+Since the sub-entity is more of a link to a resource than a representation of one a `link`  with the `self` relation doesn't exist. Consequently, a `link` object with `rel` equal to the `class` minus `"search"` & `"result"`, will provide a `href` to the resource.
+Example: `"class": [ "class", "section", "search", "result" ]` means that to fetch this resource the link with `"rel": [ "class", "section" ]` is the one whose `href` is needed.
+
+## Example Representation
+```json
+{
+    "class": [ "collection" ],
+    "entities": [
+        {
+            "class": [ "course", "search", "result" ],
+            "properties": {
+                "id": 1,
+                "name": "Software Laboratory"
+            },
+            "rel": [ "item" ],
+            "links": [
+                {
+                    "rel": [ "course" ],
+                    "href": "/v0/courses/1"
+                }
+            ]
+        },
+        {
+            "class": [ "course", "search", "result" ],
+            "properties": {
+                "id": 4,
+                "name": "Hardware Laboratory"
+            },
+            "rel": [ "item" ],
+            "links": [
+                {
+                    "rel": [ "course" ],
+                    "href": "/v0/courses/4"
+                }
+            ]
+        },
+        {
+            "class": [ "class", "search", "result" ],
+            "properties": {
+                "id": 56,
+                "name": "Hardware Laboratory 1920i"
+            },
+            "rel": [ "item" ],
+            "links": [
+                {
+                    "rel": [ "class" ],
+                    "href": "/v0/courses/4/classes/1920i"
+                }
+            ]
+        },{
+            "class": [ "course", "search", "result" ],
+            "properties": {
+                "id": 5,
+                "name": "Computer IT Laboratory"
+            },
+            "rel": [ "item" ],
+            "links": [
+                {
+                    "rel": [ "course" ],
+                    "href": "/v0/courses/5"
+                }
+            ]
+        },
+        {
+            "class": [ "todo", "search", "result" ],
+            "properties": {
+                "id": 545,
+                "name": "Hardware Laboratory - First assignment"
+            },
+            "rel": [ "item" ],
+            "links": [
+                {
+                    "rel": [ "todo" ],
+                    "href": "/v0/courses/4/21D/calendar/545"
+                }
+            ]
+        }
+    ],
+    "links": [
+        {
+            "rel": [ "next" ],
+            "href": "/v0/search?limit=5&page=2"
```

Comment:
The results seem random, since no further restrictions were applied on the query string (i.e. types and query). What will `GET "/v0/search?limit=5&page=0"` search for, by default?

Reply:
Yeah forgot the query part.